impls/src/adapters/compress.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright 2023 The Epic Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::io::{Read, Write};
+
+extern crate flate2;
+use flate2::{
+	read::{DeflateDecoder, GzDecoder, ZlibDecoder},
+	write::{DeflateEncoder, GzEncoder, ZlibEncoder},
+	Compression,
+};
+
+/// Enum to specify the desired compression format
+#[derive(Debug, Eq, Hash, PartialEq, Clone)]
+pub enum CompressionFormat {
+	// Zlib method based on flate2
+	Zlib,
+	// Glib method based on flate2
+	Gzip,
+	// Deflate method based on flate2
+	Deflate,
+}
+
+/// Transform CompressionFormat into a Vec with all options
+impl Into<Vec<CompressionFormat>> for CompressionFormat {
+	fn into(self) -> Vec<CompressionFormat> {
+		vec![
+			CompressionFormat::Zlib,
+			CompressionFormat::Gzip,
+			CompressionFormat::Deflate,
+		]
+	}
+}
+
+/// Transform into a string version of CompressionFormat
+impl ToString for CompressionFormat {
+	fn to_string(&self) -> String {
+		match *self {
+			CompressionFormat::Zlib => "zlib".to_owned(),
+			CompressionFormat::Gzip => "gzip".to_owned(),
+			CompressionFormat::Deflate => "deflate".to_owned(),
+		}
+	}
+}
+
+/// The function that handles the compression code, for now we have Zlib, Gzip and Deflate
+///  # Example
+/// ```
+/// let data = "data to compress".as_bytes();
+/// let compressed = compress(data, CompressionFormat::Gzip);
+/// ```
+pub fn compress(data: &[u8], mode: CompressionFormat) -> Vec<u8> {
+	// A vector to store the output
+	let mut output = Vec::new();
```

Comment:
Rust is throwing a warning in this line, you can apply the following to clean it:
```suggestion
	let output: Vec<u8>;
```